a3p-integration/proposals/z:acceptance/test-lib/psm-lib.js

```diff
@@ -356,46 +366,125 @@ export const initializeNewUser = async (name, fund, io) => {
 };
 
 /**
- * Similar to https://github.com/Agoric/agoric-3-proposals/blob/422b163fecfcf025d53431caebf6d476778b5db3/packages/synthetic-chain/src/lib/commonUpgradeHelpers.ts#L123-L139
- * However, in the case where "address" is not provisioned "agoric wallet send" is needed because
- * "agops perf satisfaction" tries to follow ":published.wallet.${address}" which blocks the execution because no such path exists in
- * vstorage. In situations like this "agoric wallet send" seems a better choice as it doesn't depend on following user's vstorage wallet path
+ * Similar to
+ * https://github.com/Agoric/agoric-3-proposals/blob/422b163fecfcf025d53431caebf6d476778b5db3/packages/synthetic-chain/src/lib/commonUpgradeHelpers.ts#L123-L139
+ * However, for an address that is not provisioned, `agoric wallet send` is
+ * needed because `agops perf satisfaction` hangs when trying to follow
+ * nonexistent vstorage path ":published.wallet.${address}".
  *
  * @param {string} address
  * @param {Promise<string>} offerPromise
+ * @returns {Promise<SendOfferResult>}
  */
 export const sendOfferAgoric = async (address, offerPromise) => {
   const offerPath = await mkTemp('agops.XXX');
   const offer = await offerPromise;
   await fsp.writeFile(offerPath, offer);
 
-  await agoric.wallet(
-    '--keyring-backend=test',
-    'send',
-    '--offer',
-    offerPath,
-    '--from',
-    address,
-  );
+  const [settlement] = await Promise.allSettled([
+    execa({
+      all: true,
+    })`agoric wallet --keyring-backend=test send --offer ${offerPath} --from ${address} --verbose`,
+  ]);
+  return settlement.status === 'fulfilled'
+    ? settlement.value
+    : settlement.reason;
+};
+
+/**
+ * A variant of {@link sendOfferAgoric} that uses `agd` directly to e.g.
+ * control gas calculation.
+ *
+ * @param {string} address
+ * @param {Promise<string>} offerPromise
+ * @returns {Promise<SendOfferResult>}
+ */
+export const sendOfferAgd = async (address, offerPromise) => {
+  const offer = await offerPromise;
+  const networkConfig = await getNetworkConfig({ env: process.env, fetch });
+  const { chainName, rpcAddrs } = networkConfig;
+  const args = [
+    ...[`--node=${rpcAddrs[0]}`, `--chain-id=${chainName}`],
+    ...[`--keyring-backend=test`, `--from=${address}`],
+    ...['tx', 'swingset', 'wallet-action', '--allow-spend', offer],
+    ...['-ojson', '-bblock'],
```

Comment:
why the dots? If it's for grouping, I think comment lines would be clearer
```suggestion
    // chain  
    `--node=${rpcAddrs[0]}`, `--chain-id=${chainName}`,
    // account
    `--keyring-backend=test`, `--from=${address}`,
    // transaction
    'tx', 'swingset', 'wallet-action', '--allow-spend', offer,
    // output
    '-ojson', '-bblock',
```

Reply:
Updated to use `concat` in alignment with `execSwingsetTransaction`. The above style doesn't work because our lint configuration prohibits array literals in which some but not all elements share a line.